File: leetcode/242.valid-anagram.py
# @leet start
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        CODE_PT_A = ord("a")
        counts = [0] * 26

        for ch in s:
            counts[ord(ch) - CODE_PT_A] += 1

        for ch in t:
            counts[ord(ch) - CODE_PT_A] -= 1

        for count in counts:
            if count != 0:
                return False

        return True


# @leet end

File: leetcode/242.valid-anagram.js
// @leet start
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
  const counts = new Map();

  for (const ch of s) {
    counts.set(ch, (counts.get(ch) ?? 0) + 1);
  }

  for (const ch of t) {
    counts.set(ch, (counts.get(ch) ?? 0) - 1);
  }

  for (const count of counts.values()) {
    if (count) {
      return false;
    }
  }

  return true;
};
// @leet end
